source/extensions/filters/network/zookeeper_proxy/zookeeper_utils.cc

```diff
@@ -0,0 +1,53 @@
+#include "extensions/filters/network/zookeeper_proxy/zookeeper_utils.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ZooKeeperProxy {
+
+bool BufferHelper::peekInt32(Buffer::Instance& buffer, uint64_t& offset, int32_t& val) {
+  try {
+    val = buffer.peekBEInt<int32_t>(offset);
+    offset += sizeof(uint32_t);
+    return true;
+  } catch (EnvoyException& e) {
+    return false;
+  }
+}
+
+bool BufferHelper::peekBool(Buffer::Instance& buffer, uint64_t& offset, bool& val) {
+  if (buffer.length() < (offset + 1)) {
```

Comment:
Seems like you're inconsistently doing bounds checking in these helpers. Should probably do the same for `peekInt32`.